Dockerfile

```diff
@@ -1,44 +1,62 @@
-ARG PYTHON_IMAGE_VERSION=latest
-FROM python:${PYTHON_IMAGE_VERSION}-slim
-ARG GRPC_HEALTH_PROBE_VERSION=v0.4.2
+# syntax=docker/dockerfile:1
+ARG PYTHON_IMAGE_VERSION=3.7
+FROM python:${PYTHON_IMAGE_VERSION}-slim as base
+LABEL DEPLOYMENT_TYPE="APP" maintainer="support@hydrosphere.io"
+
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    POETRY_PATH=/opt/poetry \
+    VENV_PATH=/opt/pysetup/.venv \
+    POETRY_VERSION=1.1.6 
+ENV PATH="$POETRY_PATH/bin:$VENV_PATH/bin:$PATH"
+
+FROM base AS build
+
+RUN apt-get update && apt-get -y install \
+    curl \
+    sudo \
+    wget && \
+    \
+    GRPC_HEALTH_PROBE_VERSION=v0.4.2 && wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
+    chmod +x /bin/grpc_health_probe && \
+    \
+    curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python && \
+    mv /root/.poetry $POETRY_PATH && \
+    python -m venv $VENV_PATH && \
+    poetry config virtualenvs.create false && \
+    poetry config experimental.new-installer false && \
+    pip install --upgrade pip && \
+    rm -rf /var/lib/apt/lists/*
 
-ENV APP_PORT=9091
-ENV MODEL_DIR=/model
+COPY poetry.lock pyproject.toml ./
+RUN poetry install --no-interaction
 
-ENV POETRY_HOME="/opt/poetry"
-ENV VENV_PATH="/opt/pysetup/.venv"
-ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
 
-VOLUME /model
-LABEL DEPLOYMENT_TYPE=APP
+FROM base as runtime
 
-RUN apt-get update && \
-    apt-get -y install wget curl sudo
+RUN useradd -u 42069 --create-home --shell /bin/bash app
 
-RUN wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
-    chmod +x /bin/grpc_health_probe
-RUN curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python
+WORKDIR /home/app
 
+ENV APP_PORT=9091
+EXPOSE ${APP_PORT}
 HEALTHCHECK --start-period=10s CMD /bin/grpc_health_probe -addr=:${APP_PORT}
 
-WORKDIR /app
+COPY --from=build --chown=app:app /bin/grpc_health_probe /bin/grpc_health_probe
+COPY --from=build --chown=app:app $VENV_PATH $VENV_PATH
 
-COPY poetry.lock pyproject.toml ./
-RUN poetry config virtualenvs.create false
-RUN poetry install --no-interaction
+ENV MODEL_DIR=/model
 
-COPY . /app/
+VOLUME /model
```

Comment:
Why do we need to add this mount point? We don't need persistence volume declaration into docker spec, if you want persistence in the model folder, you can mount when running the container.